src/tests/ftest/erasurecode/rebuild_fio.py

```diff
@@ -28,33 +30,69 @@ def execution(self, rebuild_mode):
         Args:
             rebuild_mode (str): On-line or off-line rebuild mode
         """
-        # Kill last server rank first
+        aggregation_threshold = self.params.get("threshold", "/run/pool/aggregation/*")
+        aggregation_timeout = self.params.get("aggr_timeout", "/run/pool/aggregation/*")
+        # 1. Disable aggregation
+        self.log_step("Disable aggregation")
+        self.pool.disable_aggregation()
+
+        # 2.a Kill last server rank first
+        self.log_step("Start fio and kill the last server")
         self.rank_to_kill = self.server_count - 1
-
         if 'on-line' in rebuild_mode:
             # Enabled on-line rebuild for the test
             self.set_online_rebuild = True
-
-        # Write the Fio data and kill server if rebuild_mode is on-line
+        # 2.b Write the Fio data and kill server if rebuild_mode is on-line
         self.start_online_fio()
 
-        # Verify Aggregation should start for Partial stripes IO
-        if not any(self.check_aggregation_status(attempt=60).values()):
-            self.fail("Aggregation failed to start..")
-
+        # 3. Get initial total free space (scm+nvme)
+        self.log_step("Get initial total free space (scm+nvme)")
+        init_free_space = self.pool.get_total_free_space(refresh=True)
+
+        # 4. Enable aggregation
+        self.log_step("Enable aggregation")
+        self.pool.enable_aggregation()
+
+        # 5. Get total space consumed (scm+nvme) after aggregation enabled, verify and wait until
+        #    aggregation triggered, maximum 3 minutes.
+        self.log_step("Verify the Fio write finish without any error")
+        start_time = time.time()
+        timed_out = False
+        aggr_triggered = False
+        self.log_step("Verify and wait until aggregation triggered")
+        while not aggr_triggered and not timed_out:
+            # Check if current free space exceeds threshold
+            free_space = self.pool.get_total_free_space(refresh=True)
+            difference = free_space - init_free_space
+            aggr_triggered = difference >= aggregation_threshold
```

Comment:
What if aggregation is never triggered? Do we fail the test?

Reply:
If aggregation did not triggered within timed_out, test will be failed.

Reply:
yes. Test will be failed.